afs/afs-core/src/main/java/com/powsybl/afs/ProjectFileCreationContext.java

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright (c) 2017, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.afs;
+
+import com.powsybl.afs.storage.ListenableAppStorage;
+import com.powsybl.afs.storage.NodeInfo;
+
+import java.util.Objects;
+
+/**
+ * @author Geoffroy Jamgotchian <geoffroy.jamgotchian at rte-france.com>
+ */
+public class ProjectFileCreationContext {
+
+    public ProjectFileCreationContext(NodeInfo info, ListenableAppStorage storage, NodeInfo projectInfo, AppFileSystem fileSystem) {
+        this.info = Objects.requireNonNull(info);
+        this.storage = Objects.requireNonNull(storage);
+        this.projectInfo = Objects.requireNonNull(projectInfo);
+        this.fileSystem = Objects.requireNonNull(fileSystem);
+    }
+
+    private final NodeInfo info;
+
+    private final ListenableAppStorage storage;
+
+    private final NodeInfo projectInfo;
+
+    private final AppFileSystem fileSystem;
```

Comment:
Why members are declared after constructor? This class is similar to ProjectFileBuildContext (refactor?)

Reply:
Fixed for 1er question + common context base class added